setup.py

```diff
@@ -309,10 +317,49 @@ def find_eigen_dir(output=False):
                 print("\n  ", dir, "  (yes)")
             return dir
         if output: print("  (no)")
+
     if output:
-        print("Could not find Eigen.  Make sure it is installed either in a standard ")
-        print("location such as /usr/local/include, or the installation directory is either in ")
-        print("your C_INCLUDE_PATH or EIGEN_DIR environment variable.")
+        print("Could not find Eigen in any of the standard locations.")
+        print("Will now try to download it from bitbucket.org. This requires an internet")
+        print("connection, so it will fail if you are currently offline.")
+        print("If Eigen is installed in a non-standard location, and you want to use that")
+        print("instead, you should make sure the right directory is either in your")
+        print("C_INCLUDE_PATH or specified in an EIGEN_DIR environment variable.")
+
+    try:
+        dir = 'downloaded_eigen'
+        if os.path.isdir(dir):
+            # If this exists, it was tried above and failed.  Something must be wrong with it.
+            shutil.rmtree(dir)
+        os.mkdir(dir)
+        url = 'https://bitbucket.org/eigen/eigen/get/3.3.4.tar.bz2'
+        if output:
+            print("Downloading eigen from ",url)
+        u = urlopen(url)
+        fname = 'eigen.tar.bz2'
+        block_sz = 32 * 1024
+        with open(fname, 'wb') as f:
+            buffer = u.read(block_sz)
+            while buffer:
+                f.write(buffer)
+                buffer = u.read(block_sz)
```

Comment:
Why the buffering?  More stable?

Reply:
I just copied this from galsim_download_cosmos, where the buffering is probably helpful for such a large file.  (And there, I'm sure I just took something from some StackOverflow answer.)  It's probably not needed here though, since I think this tarball is pretty small.  I don't think it hurts though.